retrieve/retrieve.go

```diff
@@ -0,0 +1,99 @@
+package retrieve
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/joshuatcasey/collections"
+	"github.com/paketo-buildpacks/libdependency"
+	"github.com/paketo-buildpacks/libdependency/versionology"
+	"github.com/paketo-buildpacks/libdependency/workflows"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"golang.org/x/exp/slices"
+)
+
+type GenerateMetadataFunc func(version versionology.HasVersion) (cargo.ConfigMetadataDependency, error)
+
+func NewMetadata(id string, getNewVersions libdependency.HasVersionsFunc, generateMetadata GenerateMetadataFunc, targets ...string) {
+	var (
+		buildpackTomlPath      string
+		metadataFile           string
+		targetsFile            string
+		buildpackTomlPathUsage = "full path to the buildpack.toml file, using only one of camelCase, snake_case, or dash_case"
+		metadataFileUsage      = "output filename into which to write the JSON metadata, using only one of camelCase, snake_case, or dash_case"
+		targetsFileUsage       = `output filename into which to write a JSON array of targets (e.g. ["bionic","jammy"])`
+	)
+
+	flag.StringVar(&buildpackTomlPath, "buildpackTomlPath", "", buildpackTomlPathUsage)
+	flag.StringVar(&buildpackTomlPath, "buildpack_toml_path", buildpackTomlPath, buildpackTomlPathUsage)
+	flag.StringVar(&buildpackTomlPath, "buildpack-toml-path", buildpackTomlPath, buildpackTomlPathUsage)
+	flag.StringVar(&metadataFile, "metadataFile", "", metadataFileUsage)
+	flag.StringVar(&metadataFile, "metadata_file", metadataFile, metadataFileUsage)
+	flag.StringVar(&metadataFile, "metadata-file", metadataFile, metadataFileUsage)
+	flag.StringVar(&targetsFile, "targetsFile", "", targetsFileUsage)
+	flag.StringVar(&targetsFile, "targets_file", targetsFile, targetsFileUsage)
+	flag.StringVar(&targetsFile, "targets-file", targetsFile, targetsFileUsage)
```

Comment:
What's the point of the duplicates for `-`, `_` and camel case?

Reply:
I got sick of trying to remember which command line casing was required. I'd prefer `-` everywhere if we have to pick one.